RestaurantsNearMe/Presentation/RestaurantListViewModel/RestaurantListViewModel.swift

```diff
@@ -61,24 +81,25 @@ class RestaurantListViewModel: ObservableObject {
     UIApplication.shared.open(settingsURL)
   }
   
-  func retrieveRestaurants() async {
-    await restaurantsListProvider.updateRestaurants()
-  }
-  
-  func getNextRestaurants() async {
-    // update the query
-  }
-  
-  func shouldLoadData(_ id: String) -> Bool {
-    return restaurants.firstIndex(where: { $0.id == id }) == restaurants.count - 2
+  func loadMoreDataIfNeeded(for index: Int) {
+    guard !isLoading else { return }
+    if itemsLoadedCount - index <= itemsFromEndThreshold {
+      var itemsToLoad = itemsLoadedCount
+      if let totalItemsAvailable,
+        totalItemsAvailable - itemsToLoad < 20 {
+        itemsToLoad = totalItemsAvailable - itemsToLoad
+      }
+      offsetProvider.currentPageOffset.send(itemsToLoad)
+    }
   }
```

Comment:
Essentially we're using the index of the current loaded object to assess if we need to load more from the result.

Not sure if this meets the best practice around pagination that you've seen.